code/modules/projectiles/guns/energy/laser.dm

```diff
@@ -138,3 +139,68 @@
 
 /obj/item/gun/energy/laser/redtag/hitscan
 	ammo_type = list(/obj/item/ammo_casing/energy/laser/redtag/hitscan)
+
+
+//////////////EXTERNAL LENS////////////////////////////
+
+/obj/item/external_lens
+	name = "external lens"
+	icon_state = "external"
+	desc = "These lens modify the laser frequency and trajectory giving them special effects."
+	icon = 'icons/obj/guns/energy.dmi'
+	var/stored_ammo_type = /obj/item/ammo_casing/energy/laser
+	var/restricted_type = /obj/item/gun/energy/laser //which type of laser gun it can be assigned to
+	var/overlay = "laser"
+
+/obj/item/external_lens/Initialize()
+	. = ..()
+	add_overlay(overlay)
+
+/obj/item/external_lens/afterattack(atom/movable/AM, mob/user, flag)
+	. = ..()
+	if(user && istype(AM, restricted_type) && AM.type == restricted_type)

```

Comment:
`istype(AM, restricted_type) && AM.type == restricted_type`
The istype accomplishes what you want here, as it is this makes it so it cant be applied to subtypes

Reply:
subtypes were affected with only istype()